internal/component/prometheus/relabel/relabel.go

```diff
@@ -230,7 +264,13 @@ func (c *Component) Update(args component.Arguments) error {
 	defer c.mut.Unlock()
 
 	newArgs := args.(Arguments)
-	c.clearCache(newArgs.CacheSize)
+
+	// todo maybe recreate whole relabelCache here in case of change for redis/memcached client
```

Comment:
I think this would be a no-op, generally this is used to reset the cache size. In the case if redis/memcache I imagine this would not even be called. IE the cache size should not exist within Alloy.